pkg/network/public/public.go

```diff
@@ -382,6 +388,27 @@ func EnsurePublicSetup(nodeID pkg.Identifier, vlan *uint16, inf *pkg.PublicConfi
 	return br, netlink.LinkSetUp(br)
 }
 
+func ensureTestNamespace(publicBrdige *netlink.Bridge) error {
+	netNS, err := namespace.GetByName(testNamespace)
+	if errors.Is(err, os.ErrNotExist) {
+		netNS, err = namespace.Create(testNamespace)
+		if err != nil {
+			return fmt.Errorf("failed to create namespace %s: %w", testNamespace, err)
+		}
+	}
+	if err != nil {
```

Comment:
this is probably what u want. since u already check the error of the create

```suggestion
	} else if err != nil {
```